uncompyle6/parsers/reducecheck/ifelsestmt.py

```diff
@@ -240,6 +259,34 @@ def ifelsestmt(self, lhs, n, rule, tree, tokens, first, last):
                     return True
 
             if first_offset > jmp_target:
+                # A backward or loop jump from the end of an "else"
+                # clause before the beginning of the "if" test is okay
+                # only if we are trying to match or reduce an "if"
+                # statement of the kind that can occur only inside a
+                # loop construct.
+                if lhs in ("ifelsestmtl", "ifelsestmtc"):
+                    jump_false = jmp
+                    if (
+                        tree[2] == "JUMP_FORWARD"
+                        and jump_false == "jmp_false"
+                        and len(else_suite) == 1
+                    ):
+                        suite_stmts = else_suite[0]
+                        continue_stmt = suite_stmts[0]
+                        if(
+                            suite_stmts == "suite_stmts"
+                            and len(suite_stmts) == 1
```

Comment:
There has been something nagging me about this. I suspect that rather than checking that there is only _one_ statement and it is `continue` we should be checking on the _last_ statement  being `continue`.

Reply:
`tokens[last]`?

Reply:
Somewhere or at sometime you mentioned using `tokens[last]` to look for `continue`. 

While there are many ways to do things. A potential problem with `tokens[last]` is that it might have fabricated tokens like "COME_FROM".   From experience, the preferred way to do this would be to use `suite_stmts.last_child()`.

Reply:
Yeah I removed my comment about `tokens[last]` because it was stupid, thinking about it.
`tokens` contains all the instructions from the sequence; what you meant was getting the last token _from the `else` block_.

I hadn't come across that handy `last_child()` helper function, and I'll be sure to try and remember that when/if I need it! Thanks for that.